### ISSUE TITLE
AirPlay 2 testing with 3rd party receivers

### ISSUE THREAD

**Issue by postlund** [postlund/pyatv#2035]
### What do you need help with?

I'm currently working on support for audio streaming over AirPlay v2 and I will soon need beta testers for this. In regards to what pyatv already supports (AirPlay v1/RAOP), this makes little difference when using Apple devices as they all support AirPlay v1. Adding support for v2 will hopefully make it possible to stream to (newer) receivers by 3rd party manufacturers, like portable speakers or televisions.

I have already verified my implementation to work with Apple TV 4K, AirPort Express and HomePod mini. So I'm pretty sure Apple products are covered (=need to help here). If you happen to own any other devices supporting AirPlay 2 and want to help me out testing, please let me know in this issue. I'll post a PR whenever I have something ready for testing, which should be pretty soon I hope.

**Comment by postlund** [comment]
I did manage to finish what I started and opened #2036. If anyone has some time over to try it out, that would be great!

My expectation is that pairing isn't needed on most devices, so just streaming like this should work:

    atvremote --scan-protocols airplay -s <ip> --debug stream_file=https://file-examples.com/storage/fef677cdf46481c8d96f8cd/2017/11/file_example_MP3_1MG.mp3

Any problems? Re-run with `--debug` and include the logs,

**Comment by michalmo** [comment]
Works for me with a Sonos if I comment out this line https://github.com/postlund/pyatv/blob/e57b6d392b80202739a4c65ee31b285ae50acf19/pyatv/protocols/raop/__init__.py#L399, otherwise the set_volume call fails with a 500 response

```
Traceback (most recent call last):
  File "/Users/michal/src/pyatv/pyatv/scripts/atvremote.py", line 748, in _run_application
    return await cli_handler(loop)
  File "/Users/michal/src/pyatv/pyatv/scripts/atvremote.py", line 523, in cli_handler
    return await _handle_commands(args, config, loop)
  File "/Users/michal/src/pyatv/pyatv/scripts/atvremote.py", line 636, in _handle_commands
    ret = await _handle_device_command(args, cmd, atv, loop)
  File "/Users/michal/src/pyatv/pyatv/scripts/atvremote.py", line 686, in _handle_device_command
    return await _exec_command(atv.stream, cmd, True, *cmd_args)
  File "/Users/michal/src/pyatv/pyatv/scripts/atvremote.py", line 715, in _exec_command
    value = await tmp(*args)
  File "/Users/michal/src/pyatv/pyatv/core/facade.py", line 403, in stream_file
    await self.relay("stream_file")(file, metadata=metadata, **kwargs)
  File "/Users/michal/src/pyatv/pyatv/protocols/raop/__init__.py", line 399, in stream_file
    await self.audio.set_volume(self.audio.volume)
  File "/Users/michal/src/pyatv/pyatv/protocols/raop/__init__.py", line 309, in set_volume
    await raop.set_volume(dbfs_volume)
  File "/Users/michal/src/pyatv/pyatv/protocols/raop/stream_client.py", line 408, in set_volume
    await self.rtsp.set_parameter("volume", str(volume))
  File "/Users/michal/src/pyatv/pyatv/support/rtsp.py", line 196, in set_parameter
    return await self.exchange(
  File "/Users/michal/src/pyatv/pyatv/support/rtsp.py", line 275, in exchange
    resp = await self.connection.send_and_receive(
  File "/Users/michal/src/pyatv/pyatv/support/http.py", line 473, in send_and_receive
    raise exceptions.HttpError(
pyatv.exceptions.HttpError: RTSP/1.0 method SET_PARAMETER failed with code 500: Internal Server Error
```

But set volume is failing for the Sonos through the proxy too, so maybe you just need a try/except there.

**Comment by postlund** [comment]
@michalmo Thanks for verifying 👍  So the call fails when you play from an iOS device via the proxy and change volume?

**Comment by michalmo** [comment]
It fails if I change volume from the iOS device while it's not playing, since the proxy doesn't support playing.

Testing now without proxy, if nothing is playing the volume in the Sonos app doesn't change when changing volume through iOS UI. When something is playing the volume does update in the Sonos app. So my guess is that it always responds with 500 if nothing is playing.

**Comment by postlund** [comment]
@bdraco Sorry for hi-jacking you, but I believe you have a few AirPlay 2 speakers, right? If I can get some more feedback on this and have it merged, then we can lift the Zeroconf filter we have in home Assistant and have any AirPlay 2 receiver work with TTS, which would be cool.

**Comment by postlund** [comment]
You'll need to use the PR (#2036).

**Comment by bdraco** [comment]
Testing with that PR now.
I tried with master first and got
```
2023-06-13 11:01:31.660 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140386065644560] not authenticated
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 205, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1912, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1952, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 226, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 811, in entity_service_call
    future.result()  # pop exception if have
    ^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1081, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 851, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/apple_tv/media_player.py", line 300, in async_play_media
    await self.atv.stream.stream_file(media_id)
  File "/usr/local/lib/python3.11/site-packages/pyatv/core/facade.py", line 404, in stream_file
    await self.relay("stream_file")(file, metadata=metadata, **kwargs)
  File "/usr/local/lib/python3.11/site-packages/pyatv/protocols/raop/__init__.py", line 357, in stream_file
    await client.initialize(self.core.service.properties)
  File "/usr/local/lib/python3.11/site-packages/pyatv/protocols/raop/raop.py", line 457, in initialize
    await self._setup_session()
  File "/usr/local/lib/python3.11/site-packages/pyatv/protocols/raop/raop.py", line 486, in _setup_session
    resp = await self.rtsp.setup(
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pyatv/support/rtsp.py", line 179, in setup
    return await self.exchange("SETUP", headers=headers, body=body)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pyatv/support/rtsp.py", line 263, in exchange
    resp = await self.connection.send_and_receive(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pyatv/support/http.py", line 434, in send_and_receive
    raise exceptions.AuthenticationError("not authenticated")
pyatv.exceptions.AuthenticationError: not authenticated
```

**Comment by bdraco** [comment]
The onelink smoke detector works great with that PR

**Comment by bdraco** [comment]
Working fine on various sonos devices

**Comment by bdraco** [comment]
I tried a few models and devices.  Everything worked

**Comment by postlund** [comment]
@bdraco Perfect, thank you for the quick response! I'll try to make a release today.

@michalmo Anything you want to add before I make the release?

**Comment by michalmo** [comment]
not at the moment

**Comment by postlund** [comment]
Bug fixes are always welcome, of course! I will take a look at it tomorrow!

**Comment by postlund** [comment]
I have merged #2036 now, so it will be part of the next release 🚀
